src/deepEqual.js

```diff
@@ -19,8 +19,28 @@
  * deepEqual({test: {abc: 5}}, {test: {abc: 5}}) === true
  * deepEqual({test: {abc: 5}}, {test: {abc: 5, def: 4}}) === false
  */
-function deepEqual(a, b) {
-  // write code here
+function deepEqual(obj1, obj2) {
+  if (obj1 === obj2) {
+    return true;
+  }
+
+  if (obj1 === null || typeof (obj1) !== 'object' ||
+      obj2 === null || typeof (obj2) !== 'object') {
+    return false;
+  }
+
+  var propertiesInObj1 = 0;
+  var propertiesInObj2 = 0;
+  for (var property in obj1) {
+    propertiesInObj1 += 1;
+  }
```

Comment:
you can do 
numPropsInObj1 = obj1.keys().length 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys

Reply:
why not
numPropertiesInObj1 = Object.keys(obj1).length ?